server/src/main/java/org/opensearch/transport/Header.java

```diff
@@ -60,11 +60,11 @@ public class Header {
     private final long requestId;
     private final byte status;
     // These are directly set by tests
-    String actionName;
-    Tuple<Map<String, String>, Map<String, Set<String>>> headers;
+    public String actionName;
```

Comment:
@VachaShah could you please revert changes in this file and move `InboundDecoder` and `InboundAggregator` back to `transport` package (at least for now)?

Yes, that would impact tests but we need to be a little bit more creative at the moment in order to contain the change.

For tests, the idea is very simple (we would need to move back `InboundDecoderTests`, `InboundPipelineTests` and `InboundAggregatorTests`, and probably `NativeOutboundHandlerTests` as well). If you look at `InboundDecoderTests` and `InboundPipelineTests`, those do not need access to message at all, so we could make these classes abstract and implement for each protocol, for example:

```java
public abstract class InboundDecoderTests extends OpenSearchTestCase {

    protected abstract BytesReference serialize(boolean isRequest, Version version, boolean handshake, boolean compress, 
          String action, long requestId) throws IOException;

     // all tests stay there

     public void testDecode() throws Exception { 
         // ...
         // Instead of creating NativeOutboundMessage directly, we use `serialize` method, that each protocol would implement
         final BytesReference totalBytes = serialize(isRequest, Version.CURRENT, false, false, action, requestId);
          // ...
     }
}
```


```java
package org.opensearch.transport.nativeprotocol;

import org.opensearch.Version;
import org.opensearch.common.io.stream.BytesStreamOutput;
import org.opensearch.core.common.bytes.BytesReference;
import org.opensearch.transport.InboundDecoderTests;
import org.opensearch.transport.TestRequest;
import org.opensearch.transport.TestResponse;

import java.io.IOException;
import java.util.Collections;

public class NativeInboundDecoderTests extends InboundDecoderTests {
    @Override
    protected BytesReference serialize(boolean isRequest, Version version, boolean handshake, boolean compress, String action, long requestId) throws IOException {
        NativeOutboundMessage message;
        if (isRequest) {
            message = new NativeOutboundMessage.Request(
                threadContext,
                new String[0],
                new TestRequest(randomAlphaOfLength(100)),
                version,
                action,
                requestId,
                handshake,
                compress
            );
        } else {
            message = new NativeOutboundMessage.Response(
                threadContext,
                Collections.emptySet(),
                new TestResponse(randomAlphaOfLength(100)),
                version,
                requestId,
                handshake,
                compress
            );
        }

        final BytesReference totalBytes = message.serialize(new BytesStreamOutput());
        return totalBytes;
    }

}

```